### PR TITLE
Chain-able OutlineEffect

### PR DESCRIPTION
**Pull request by takahirox**
This PR enables chain-able `*Effect`.

`*Effect` which inherits `ChainableEffect` can work under other `*Effect`
and two or more `*Effect` can work at a time.

For example: using `VREffect` and `OutlineEffect`

```
function init() {

    var renderer = new THREE.WebGLRenderer();
    var effect = new THREE.VREffect( new THREE.OutlineEffect( renderer ) );

}

function render() {

    effect.render( scene, camera );

}
```

image

(Model: Tda's Miku append http://seiga.nicovideo.jp/seiga/im2018614)

I've applied `ChainableEffect` only to `OutlineEffect` because
it seems that other `*Effect`s don't need this ability so far.

I have a plan to make chain-able `ToonMapEffect` once `toonMap` PR is berged
and users can easily do cel shading with `OutlineEffect` and `ToonMapEffect`.

And I've removed `render()` from `MMDHelper` because
it doesn't need to do anything special for `*Effect` now.

### CODE REVIEW

**Comment by mrdoob** [comment]
I think the approach in this PR is a bit hard to follow. How about...

```js
THREE.Effect = function ( renderer ) {

	this.setSize = function ( width, height ) {

		renderer.setSize( width, height );

	};
	
	this.render = function ( scene, camera ) {

		renderer.render( scene, camera );

	};

};

THREE.OutlineEffect = function ( renderer, parameters ) {
  
	THREE.Effect.call( this, renderer );

	...

}
```

**Comment by takahirox** [comment]
I had two ideas. One is in this PR. Another one is what you suggested.

I chose the first one because I don't wanna let `ChainableEffect` suffer from `WebGLRenderer` APIs change.
But the second one is also nice because probably `WebGLRenderer` rarely changes its APIs.
And `ChainableEffect` will be more straightforward.
I'll make wrapper methods existing `*Effect` uses so far.
Adding other wrapper methods when they're required won't be troublesome so much I think.

BTW about  the classname, I wanna name `ChainableEffect` or something rather than `Effect`
because this class aim is just to give "being called from other `*Effect`" ability which
only some of `*Effect`s need.

I mean I don't wanna bother `VREffect` users to add `<script src="effects/Effect.js></script>`
which is unnecessary for them in their code.

What do you think?

**Comment by mrdoob** [comment]
Hmm... I'm thinking that something like `renderer.effects = [ new THREE.OutlineEffect(), new THREE.VREffect() ]; will result in a more understandable API.

We can, internally, still do the chaining somehow.

**Comment by takahirox** [comment]
Without changing existing `*Effect`s?

**Comment by takahirox** [comment]
I've changed the aim of this PR from general `ChainableEffect` to chain-able `OutlineEffect`.

I think to achieve chain-able `renderer.effects = [ new THREE.OutlineEffect(), new THREE.VREffect() ];` requires not small changes and it takes time.

It's a good idea to focus on only `OutlineEffect` so far and we can keep discussing for general  `renderer.effects`.

File: examples/js/effects/OutlineEffect.js [hunk 1]
```diff
@@ -26,11 +26,49 @@
...
+			this[ key ] = renderer[ key ].bind( renderer );
+
+		} else {
+
+			// just copy property
+			this[ key ] = renderer[ key ];
+
+		}
+
+	}
```
Comment (takahirox) [inline examples/js/effects/OutlineEffect.js]: I added some comments, but still hard to follow?
Should I replace that with

    this.setSize = function ( width, height ) {

        renderer.setSize( width, height );

    };

    this.getSize = function () {

        return renderer.getSize();

    };

    ....

?

Comment (mrdoob) [inline examples/js/effects/OutlineEffect.js]: I think so, yes.

Comment (takahirox) [inline examples/js/effects/OutlineEffect.js]: OK, I see!

**Comment by takahirox** [comment]
I've updated!

**Comment by mrdoob** [comment]
Ops! Conflicts...

**Comment by mrdoob** [comment]
Thanks!
